modeltranslation_wagtail/patch_wagtailadmin.py

```diff
@@ -0,0 +1,669 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, unicode_literals
+
+import copy
+import logging
+import operator
+import inspect
+from six import iteritems
+try:
+    from functools import reduce
+except ImportError:
+    pass
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.urlresolvers import reverse
+from django.db import transaction
+from django.db.models import Q
+from django.http import Http404
+from django.utils.translation import ugettext as _
+
+from wagtail.wagtailadmin.edit_handlers import FieldPanel, \
+    MultiFieldPanel, FieldRowPanel
+from wagtail.wagtailadmin.edit_handlers import StreamFieldPanel
+from wagtail.wagtailcore.models import Page, Site
+from wagtail.wagtailcore.url_routing import RouteResult
+from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
+from wagtail.wagtailsearch.index import SearchField
+from wagtail.wagtailsnippets.views.snippets import get_snippet_edit_handler, \
+    SNIPPET_EDIT_HANDLERS
+try:
+    from wagtail.wagtailadmin.views.pages import get_page_edit_handler, \
+        PAGE_EDIT_HANDLERS
+except ImportError:
+    pass
+
+
+from modeltranslation.translator import translator, NotRegistered
+from modeltranslation.utils import build_localized_fieldname
+
+from .patch_wagtailadmin_forms import WagtailModeltranslationAdminPageForm
+
+logger = logging.getLogger('wagtail.core')
+
+
+class WagtailTranslator(object):
+    _patched_models = []
+
+    def __init__(self, model):
+
+        # Check if this class was already patched
+        if model in WagtailTranslator._patched_models:
+            return
+
+        WagtailTranslator._base_model = model
+        WagtailTranslator._required_fields = {}
```

Comment:
I had a chance now to try and merge this, but it's impossible. It's just one gigantic mess :/ For instance, here we unconditionally set a class type property when creating a new instance. Then later access the dictionary and hit a KeyError because `WagtailTranslator._required_fields[model]` doesn't exist... big surprise when it's reset every time a new instance is generated.

I don't know how this code was conceived, but it's quite crazy and impossible to clean up for me.

If it just worked, that would be cool... but even so, this project would become as unmaintainable as wagtail-modeltranslation :)